mods/persistence/controllers/subsystems/persistence.dm

```diff
@@ -161,13 +158,37 @@ SUBSYSTEM_DEF(persistence)
 					CHECK_TICK
 			serializer.Commit() // cleanup leftovers.
 
-		for(var/AK in areas_to_save)
-			var/area/A = areas_to_save[AK]
-			var/z
+		index = 1
+
+		// Repeat much of the above code in order to save areas marked to be saved that are not in a saved z-level.
+		for(var/area/A in saved_areas)
 			for(var/turf/T in A)
-				z = T.z
-				break
-			serializer.Serialize(A, null, z)
+				if(T.z in saved_levels)
+					continue
+				var/turf/default_turf = get_base_turf(T.z)
+				if(!istype(T) || istype(T, default_turf))
+					if(!istype(T) || !T.contents || !length(T.contents))
+						continue
+					var/should_skip = TRUE
+					for(var/atom/movable/AM in T.contents)
+						if(AM.simulated && AM.should_save())
+							should_skip = FALSE
+							break // We found a thing that's worth saving.
+					if(should_skip)
+						continue // Skip this tile. Not worth saving.
+				serializer.Serialize(T, null, T.z)
+
+				// Don't save every single tile.
+				// Batch them up to save time.
+				if(index % 128 == 0)
+					serializer.Commit()
+					index = 1
+				else
+					index++
+
+				// Prevent the whole game from locking up.
+				CHECK_TICK
+			serializer.Commit() // cleanup leftovers.
```

Comment:
This particular section needs to be wholly reviewed and likely replaced, as it's almost a 1:1 copy paste from the serialization code above.